closure-deps/lib/normalize.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * @license
+ * Copyright 2022 The Closure Library Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const path = require('path');
+
+/**
+ * Escape regular expression pattern.
+ * @param {string} pattern 
+ * @returns {string}
+ */
+function escapePattern(pattern) {
+  return pattern.replace(/\\/g, '\\\\');
+}
+
+/**
+ * Replaces OS-dependent path with POSIX style path.
+ * @param {string} p 
+ * @returns {string}
+ */
+exports.normalizePath = (p) => {
+  return p.replace(new RegExp(escapePattern(`^${path.resolve('/')}`)), path.posix.sep)
```

Comment:
This is a lot of nested parentheses, making it difficult to understand what's happening.  This could be clearer as

```
const root = new RegExp(...);
const nonPosixSep = new RegExp(...);
return p.replace(root, path.posix.sep).replace(nonPosixSep, path.posix.sep);
```
```

Reply:
I changed it almost as suggested.
Only one point, the second regular expression matches the POSIX style separator on systems that use POSIX style paths, so I changed the variable name.